src/rgw/setup/rgw.py

```diff
@@ -710,52 +706,35 @@ def _update_hax_endpoint_and_create_admin(conf: MappedConf):
-
-            #    data_pod_hostnames.remove(current_data_node)
-            for data_pod_hostname in data_pod_hostnames:
-                try:
-                    Rgw._update_hax_endpoint(conf, data_pod_hostname)
-                except SetupError as e:
-                    Log.debug(f'Error occured while updating hax endpoints. {e}')
-                    continue
-                status = Rgw._create_rgw_user(conf)
-                if status == 0:
-                    Log.info(f'User creation is successful on "{Rgw._machine_id}" node.')
-                    Rgw._set_consul_kv(rgw_consul_idx, const.CONSUL_LOCK_KEY, const.ADMIN_USER_CREATED)
-                    break
-                else:
-                    if data_pod_hostname == data_pod_hostnames[-1]:
-                        Rgw._delete_consul_kv(rgw_consul_idx, const.CONSUL_LOCK_KEY)
-                        raise SetupError(status, 'Admin user creation failed on'
-                            f' "{Rgw._machine_id}" node, with all data pods - {data_pod_hostnames}')
+            data_pod_hostname = Rgw._get_random_data_node(conf)
+            try:
+                Rgw._update_hax_endpoint(conf, data_pod_hostname)
+            except SetupError as e:
+                raise SetupError(errno.EINVAL, f'Error occured while updating hax endpoints. {e}')
+            # create admin user
+            status = Rgw._create_rgw_user(conf)
+            if status == 0:
+                Log.info(f'User creation is successful on "{Rgw._machine_id}" node.')
+                Rgw._set_consul_kv(rgw_consul_idx, const.CONSUL_LOCK_KEY, const.ADMIN_USER_CREATED)
+            else:
+                Rgw._delete_consul_kv(rgw_consul_idx, const.CONSUL_LOCK_KEY)
+                raise SetupError(status, 'Admin user creation failed on'
+                    f' "{Rgw._machine_id}" node with data pod endpoint - {data_pod_hostname}')
 
 
     @staticmethod
-    def _get_data_nodes(conf: MappedConf):
-        """Return all data nodes hostname from GConf"""
-        data_pod_hostnames = []
-        Log.debug('Collecting all data pod hostnames from GConf..')
+    def _get_random_data_node(conf: MappedConf):
+        """Return any random data nodes hostname from GConf"""
+        Log.debug('Collecting one random data pod hostnames from GConf..')
         node_identify_keys = Rgw._search_cortx_conf(conf, const.NODE_IDENTIFIER, const.DATA_NODE_IDENTIFIER)
         node_machine_ids = list(map(lambda x: x.split('>')[1], node_identify_keys))
-        for machine_id in node_machine_ids:
-            data_pod_hostnames.append(Rgw._get_cortx_conf(conf, const.NODE_HOSTNAME % machine_id))
+        if len(node_machine_ids) <= 0:
+           raise SetupError(errno.EINVAL, 'Failed to fetch data node machine ids from Gconf !!')
 
-        Log.debug(f'Collected all data pod hostnames from GConf : {data_pod_hostnames}')
-        return data_pod_hostnames
+        random_machine_id = node_machine_ids[0]
```

Comment:
earlier we use to fetch specific data pod names which was running on same worker node as of server pod. now we don't have any such dependency hence I added it as random even though it will pick up first node name returned from Gconf node list.

Reply:
yeah, please change the name as its little misleading

Reply:
done

Reply:
done